client/js/widgets/button.js

```diff
@@ -53,12 +53,19 @@ export class Button extends Widget {
 
     if(this.p('debug'))
       $('#debugButtonOutput').textContent = '';
-
+   let thisID = this.p('id');
     const variables = {
       playerName,
-      playerColor
+      playerColor,
+      "thisID" : thisID
```

Comment:
This is actually the same as writing `thisID: thisID` which in turn can be shortened to just `thisID` (like the others).

Since you don't need `thisID` anywhere else (see comment below), I would have probably written `thisID: this.p('id')` but that's just me.